src/classes/settings.py

```diff
@@ -111,3 +125,75 @@ def save(self):
             # try to save data to file, will raise exception on failure
             self.write_to_file(file_path, self._data)
 
+    def pathSettings(self, action: actions):
+        """Given an action, return the corresponding setting names"""
+        if action == self.actions.IMPORT:
+            return {'type':"locationImportType", 'path':"locationImportPath"}
+        if action == self.actions.EXPORT:
+            return {'type': "locationExportType", 'path': "locationExportPath"}
+        if action in [self.actions.SAVE, self.actions.LOAD]:
+            return {'type': "locationProjectType", 'path': "locationProjectPath"}
+        log.info("Action not recognized")
+        log.error("Given action is not valid %s" % action)
+        return None
+
+    def setDefaultPath(self, action: pathType, recent_path: str):
+        """ Change the path setting corresponding to the given action """
+        if not os.path.exists(recent_path):
+            return
+        if os.path.isfile(recent_path):
+            recent_path=os.path.dirname(recent_path)
+        try:
+            setting = self.pathSettings(action)
+            log.debug(f"Setting %s to %s" % (action.value, recent_path))
+            self.set(setting["path"], recent_path)
+        except:
+            log.error(f"Error setting the path of %s" % setting["path"])
```

Comment:
Will `setting` even be defined at this point in the `except` clause?  Perhaps the log.error could be a safer message, that will definitely work?

Reply:
Makes sense. I'll make sure the error msg is safe